DEV_GUIDE.md

```diff
@@ -0,0 +1,166 @@
+
+```scala
+import org.scalatest._
+import org.scalaexercises.definitions.Section
+
+/** @param name section_title (1)
+*/
+object MySection extends FlatSpec with Matchers with Section (2) {
+    // exercises definition (3)
+}
+```
+
+* (1) As when defining the library, you need to use [ScalaDoc](https://wiki.scala-lang.org/display/SW/Syntax) to define the name of the section. Use an underscore symbol to include spaces.
+* (2) Each section `object` should inherit from `org.scalaexercises.definitions.Section`. `FlatSpec` and `Matchers` are traits from the ScalaTest library, the framework Scala Exercises uses to check the results.
+* (3) Inside the section `object` we can start defining the exercises, as it'll be shown in the following section.
+
+
+####Creating new exercises
+
+The content for each exercise is defined in a [ScalaDoc](https://wiki.scala-lang.org/display/SW/Syntax) block, with some specific particularities.
+
+```scala
+/** = Title = (1)
+*
+* Text describing background about the exercise, can be as long as needed. (2)
+*
+* {{{
+*   // code block (3)
+* }}}
+*
+* Documentation can be broken in as many paragraphs as needed.
+*/
+def functionAssert(res0: Boolean): Unit {
+    true shouldBe res0  // (4)
+}
+```
+
+* (1) You can define titles in the documentation for your library wrapping it inside a pair of equal signs (=), defining a [ScalaDoc](https://wiki.scala-lang.org/display/SW/Syntax) heading.
+* (2) The documentation itself can be splitted up in several paragraphs, depending on your needs. You can use backquotes (`) to highlight an specific item or a code reference.
+* (3) Code that illustrates your exercises should be surrounded by triple curly braces.
+* (4) Code that defines your exercises is contained in a Scala function below the [ScalaDoc](https://wiki.scala-lang.org/display/SW/Syntax) containing the documentation.
+
+Exercise functions (as described in (4)) should receive one parameter per placeholder that users will fill in the UI, and return `Unit`. You can include as many placeholders as needed, with a minimum of one. Inside the function, you can include nested functions that your exercise may need to work, and a set of ScalaTest asserts checking the user's inputs.
+
+Currently Scala Exercises only supports this kind of input (i.e.: filling placeholders). Potentially it'll allow writing complete code snippets from the users in order to complete an exercise.
+
+####Local development
+
+In order to see and debug your exercises in your local environment, you need to publish them locally and include them as a dependency for your local instance. Here are the steps to achieve this:
+
+* (1) Compile your exercises (i.e.: `sbt compile`). This is needed for each change, as the publication process can fail otherwise.
+* (2) Publish your excersises locally (i.e.: `sbt publishLocal`).
+* (3) Run [Scala Exercises](https://github.com/scala-exercises/scala-exercises) in your local development environment (i.e.: `sbt run`).
```

Comment:
This is missing a step about putting the dependency in `build.sbt` for `scala-exercises`? This means that before this, you should mention the fields that should go in the exercises `build.sbt` (Organization/Name/Version).